lavalink/node.py

```diff
@@ -266,10 +272,13 @@ async def wait_until_ready(self, timeout: Optional[float] = None):
         await asyncio.wait_for(self._ready_event.wait(), timeout=timeout)
 
     def _get_connect_headers(self) -> dict:
+        # Num-Shards is not used on Lavalink jar files >= v3.4
+        # but kept for compatability to avoid NPEs on older builds
         headers = {
             "Authorization": self.password,
             "User-Id": str(self.user_id),
             "Num-Shards": str(self.num_shards),
+            "Client-Name": str(self._client_name),
```

Comment:
This is an equivalent of a user agent, it should just contain the name and version of the library, see Freya's Lavalink library for an example:
https://github.com/freyacodes/Lavalink-Client/blob/8d9b660b2c87bb4e286409e1c6aabe36f81a53d8/src/main/java/lavalink/client/io/Lavalink.java#L111

[Per IMPLEMENTATION.md](https://github.com/freyacodes/Lavalink/blob/092f602b787307b78e7b78625121ba56d2723f54/IMPLEMENTATION.md#opening-a-connection), the recommended format seems to be NAME/VERSION:
```suggestion
            "Client-Name": f"Red-Lavalink/{__version__}",
```
This means that any other changes related to client name (`self._client_name`, `client_name` param) can be reverted.